assets/javascripts/discourse/components/group-reports-nav-item.js.es6

```diff
@@ -0,0 +1,21 @@
+import { ajax } from "discourse/lib/ajax";
+
+export default Ember.Component.extend({
+  group: null,
+  showReportsTab: false,
+
+  checkForReports() {
+    return ajax(`/g/${this.group.name}/reports`).then(response => {
+      return this.set("showReportsTab", response.queries.length > 0);
```

Comment:
There are lots of places in this PR that use `.set` no need really.

`this.showReportsTab = response.queries.length > 0;` reads much better (there are quite a few similar places in the PR)

Reply:
@SamSaffron I agree it is cleaner, but I kept getting errors when trying to set without calling `this.set`. I got this error when I tried to not use set.

![image](https://user-images.githubusercontent.com/16214023/64576682-4ecd7980-d33f-11e9-8dc8-9f0fc2d68cfd.png)
![image](https://user-images.githubusercontent.com/16214023/64576680-4a08c580-d33f-11e9-819b-21c3258b9ebc.png)

Reply:
:cry:  I wonder if there is a workaround here for components @eviltrout / @jjaffeux  ? Seems like a tricky rule to remember.